tests/unit/test_loaders.py

```diff
@@ -248,3 +248,100 @@ def test_genGraphFromContactFile(commute_moves):
+
+        b = loaders.AgeRange(10, 20)
+        assert a == b
+        assert 10 in b
+        assert 20 not in b
+
+        c = loaders.AgeRange((10,20))
+        assert b == c
+
+        d = loaders.AgeRange("70+")
+        assert 99 in d
+        assert 69 not in d
+
+        e = loaders.AgeRange("invalid")
+    assert e_info.value.args[0] == f'Invalid age range specified: "invalid"'
+
+    with pytest.raises(Exception) as e_info:
+        a = loaders.AgeRange("[10,5)")
+    assert e_info.value.args[0] == f'Invalid age range specified: [10,5)'
+
+    with pytest.raises(Exception) as e_info:
+        a = loaders.AgeRange(10, 5)
+    assert e_info.value.args[0] == f'Invalid age range specified: [10,5)'
+
+    with pytest.raises(Exception) as e_info:
+        a = loaders.AgeRange(20,30)
+        b = loaders.AgeRange(25,35)
+        loaders._check_overlap(a, b)
+    assert e_info.value.args[0] == f"Overlap in age ranges with {a} and {b}"
+    with pytest.raises(Exception) as e_info:
+        a = loaders.AgeRange("70+")
+        b = loaders.AgeRange("[65,75)")
+        loaders._check_overlap(a, b)
+    assert e_info.value.args[0] == f"Overlap in age ranges with {a} and {b}"
+
+
+def test_invalidMixingMatrixFiles():
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as fp:
+        rows = [
+            ',"[0,15)","[0,15)"',
+            '"[0,15)",0.1, 0.05',
+            '"[15,30)",0.2, 0.2',
+        ]
+        fp.write("\n".join(rows))
+        fp.flush()
+        with pytest.raises(Exception) as e_info:
+            loaders.MixingMatrix(fp.name)
+        assert e_info.value.args[0] == "Duplicate column header found in mixing matrix"
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as fp:
+        rows = [
+            ',"[0,15)","[15,30)"',
+            '"[0,15)",0.1, 0.05',
+            '"[0,15)",0.2, 0.2',
+        ]
+        fp.write("\n".join(rows))
+        fp.flush()
+        with pytest.raises(Exception) as e_info:
+            loaders.MixingMatrix(fp.name)
+        assert e_info.value.args[0] == "Duplicate row header found in mixing matrix"
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as fp:
+        rows = [
+            ',"[0,20)","[15,30)"',
+            '"[0,15)",0.1, 0.05',
+            '"[15,30)",0.2, 0.2',
+        ]
+        fp.write("\n".join(rows))
+        fp.flush()
+        with pytest.raises(Exception) as e_info:
+            loaders.MixingMatrix(fp.name)
+        assert e_info.value.args[0] == "Overlap in age ranges with [0,20) and [15,30)"
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as fp:
+        rows = [
+            ',"[0,15)","[15,30)"',
+            '"[0,15)",0.1, 0.05',
+            '"[10,30)",0.2, 0.2',
+        ]
+        fp.write("\n".join(rows))
+        fp.flush()
+        with pytest.raises(Exception) as e_info:
+            loaders.MixingMatrix(fp.name)
+        assert e_info.value.args[0] == "Overlap in age ranges with [0,15) and [10,30)"
```

Comment:
We have different test cases here. I think that, if we split into multiple test functions, it will be easier to see what went wrong when a test fails

- [ ] split into multiple test functions

Reply:
I split this function into two test cases - one for overlapping ranges and one for duplicate headers.